packages/shared/src/graphql/posts.ts

```diff
@@ -553,24 +553,28 @@ export interface EditPostProps {
   title: string;
   content: string;
   image: File;
+  type: PostType;
 }
 
 export interface CreatePostProps
   extends Pick<EditPostProps, 'title' | 'content' | 'image'> {
   sourceId: string;
 }
 
-export type CreatePostModerationProps = {
-  title?: string;
-  content?: string;
-  sourceId: string;
-  type: PostType;
-  sharedPostId?: string;
-  externalLink?: string;
-  imageUrl?: string;
-  image?: File;
-  commentary?: string;
-};
+export type CreatePostModerationProps =
+  | {
+      title?: string;
+      content?: string;
+      sourceId: string;
+      type: PostType;
+      sharedPostId?: string;
+      externalLink?: string;
+      imageUrl?: string;
+      image?: File;
+      commentary?: string;
+      postId?: string;
+    }
+  | { postId: string; type?: PostType; image: File };
```

Comment:
Why do we need only three props for edit?

Reply:
@sshanzel Totally forgot about this one! You're right, we should have the rest of the props, too!